src/java/org/apache/cassandra/index/sai/disk/v1/postings/PostingsReader.java

```diff
@@ -334,7 +329,10 @@ private void advanceOnePosition(long nextPosting)
     private void reBuffer() throws IOException
     {
         long pointer = summary.offsets.get(blockIndex);
-
+        if (pointer < 4) {
+            // the first 4 bytes must be CODEC_MAGIC
+            throw new CorruptIndexException(String.format("Invalid block offset %d for postings block idx %d", pointer, blockIndex), input);
+        }
```

Comment:
nit: could just blow the curly braces away on this one...it'll never be more than a single line I guess

Reply:
true, but i prefer the braces just to make it obvious that it's correct, even though the second line is just a comment

Reply:
Quick, put the first `{` on a new line!